### ISSUE TITLE
EF Core Issues with Owned Types in multiple Many-To-Many Relationships between two entities

### ISSUE THREAD

**Issue by bbartels** [dotnet/efcore#25224]
## File a bug

I was trying to create two many-to-many relationships between the same types. This does not cause any issues until a related entity has two or more owned types (it works fine with a single owned type). 

I've searched previously filed issue and could not find , however there are so many and the exception message is very generic that I might have missed it.

### Include your code

Below is the offending DBContext, I've attached a full reproduction here:  https://gist.github.com/bbartels/685bfd47a20f1e7db85f89af64b3422f

```csharp
  builder.Entity<Item>(e =>
  {
      e.HasKey(x => x.Guid);
  });

  builder.Entity<Lab>(e =>
  {
      e.HasKey(e => e.Guid);
      e.Property(x => x.OtherId).IsRequired();
      e.OwnsOne(x => x.Value1);
      e.OwnsOne(x => x.Value2);
  });

  builder.Entity<Other>(e =>
  {
      e.HasKey(x => x.Guid);
      e.HasMany(x => x.Items).WithMany("Groups1");
      e.HasMany(x => x.OtherItems).WithMany("Groups2");
      e.HasMany(x => x.Labs).WithOne().IsRequired().HasForeignKey(x => x.OtherId);
  });
```

### Include stack traces

```
  Message: 
    System.ArgumentException : An item with the same key has already been added. Key: MinRepro.ValueEntity

  Stack Trace: 
    Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
    Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
    Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
    ManyToManyJoinEntityTypeConvention.CreateJoinEntityType(IConventionSkipNavigationBuilder skipNavigationBuilder)
    ManyToManyJoinEntityTypeConvention.ProcessSkipNavigationInverseChanged(IConventionSkipNavigationBuilder skipNavigationBuilder, IConventionSkipNavigation inverse, IConventionSkipNavigation oldInverse, IConventionContext`1 context)
    ImmediateConventionScope.OnSkipNavigationInverseChanged(IConventionSkipNavigationBuilder navigationBuilder, IConventionSkipNavigation inverse, IConventionSkipNavigation oldInverse)
    OnSkipNavigationInverseChangedNode.Run(ConventionDispatcher dispatcher)
    DelayedConventionScope.Run(ConventionDispatcher dispatcher)
    ConventionBatch.Run()
    ConventionBatch.Dispose()
    <28 more frames...>
    DbContext.get_DbContextDependencies()
    DbContext.get_InternalServiceProvider()
    IServiceProvider>.get_Instance()
    InfrastructureExtensions.GetService[TService](IInfrastructure`1 accessor)
    AccessorExtensions.GetService[TService](IInfrastructure`1 accessor)
    DatabaseFacade.get_Dependencies()
    DatabaseFacade.CreateExecutionStrategy()
    RelationalDatabaseFacadeExtensions.OpenConnection(DatabaseFacade databaseFacade)
    UnitTest1.FailingTest() line 90
```

### Include verbose output

Verbose output seems to have no important information, I can include it if desired.

### Include provider and version information

EF Core version: 5.0.7
Database provider: Microsoft.EntityFrameworkCore.Sqlite
Target framework: .NET 5.0
Operating system: Windows
IDE: Visual Studio 2022 Preview 1

**Comment by smitpatel** [comment]
This is likely a bug in EF Core. Though the model configuration will not work. navigations which doesn't exist on CLR type are not supported.

**Comment by bbartels** [comment]
Yeah I am aware, I am changing the property access mode to the backing field in my actual code. I just wanted to remove as much noise as possible to make finding the issue easier!

**Comment by enashed** [comment]
I have the same issue. Is there any workaround in the model configurations ?

**Comment by bbartels** [comment]
I just configured a single many to many and dealt with separation of the two independent groups in my domain model. Not pretty, but it works.

**Comment by AndriySvyryd** [comment]
Doesn't repro anymore. Likely fixed in 762e870a2d748bab144ddc733ac899a610cc0106
